src/jsonapi_client/resourceobject.py

```diff
@@ -409,11 +409,10 @@ def _handle_data(self, data):
         self.links = Links(self.session, data.get('links', {}))
         self.meta = Meta(self.session, data.get('meta', {}))
 
-        self._attributes = AttributeDict(data=data['attributes'],
-                                         resource=self)
         self._relationships = RelationshipDict(
             data=data.get('relationships', {}),
             resource=self)
+        self._attributes = AttributeDict(data=data.get('attributes', {}), resource=self)
```

Comment:
If there is neither attributes nor relationships fields, how does this ResourceObject instance differ from a ResourceIdentifier instance?

Reply:
@divums - I'm not sure, but I think the difference is in optional fields that can be specified. I.e. a [resource](https://jsonapi.org/format/#document-resource-objects) only needs an ID and type but other fields such as attributes/relationships *can* be specified as well, whereas a [resource identifier](https://jsonapi.org/format/#document-resource-identifier-objects) can only have an ID, type and meta and so is more constrained.

In our case, we have resources without attributes but with relationships and so needed this change. 

Under the spec I don't think there is a difference between a Resource and ResourceIdentifier in terms of required fields.

Reply:
Right, there is no difference between the rendered ResourceIdentifier and ResourceObject. I was just wondering aloud if there should be an assert in ResourceObject to require other fields than `type` and `id`, otherwise fallback to ResourceIdentifier. But that doe not seem to be a good idea.